dll/win32/shell32/folders/CControlPanelFolder.cpp

```diff
@@ -773,3 +773,144 @@ HRESULT WINAPI CCPLItemMenu::HandleMenuMsg(
 
     return E_NOTIMPL;
 }
+
+/**************************************************************************
+* COpenControlPanel
+*/
+
+static HRESULT GetParsingName(PCIDLIST_ABSOLUTE pidl, PWSTR*Name)
```

Comment:
Do you expect to call `GetParsingName` on _actual_ pidls in the future? (So far the function is only used on CSIDLs, for which you are then forced to cast the CSIDL to a PCIDLIST.... pointer)

Reply:
For the regular .cpl files yes, that was the intent IIRC (IShellFolder2::GetDetailsOfEx required to handle .cpl files this way though).

Could be a SHELL32 helper function in util.cpp perhaps.